src/main/scala/com/kinja/soy/Soy.scala

```diff
@@ -63,4 +63,7 @@ object Soy {
    * Generates a SoyMapWrites[T] for a given case class T.
    */
   def writes[T]: SoyMapWrites[T] = macro SoyMacroImpl.writesImpl[T]
+
+  def typedWrites[T](typeValue: String): SoyMapWrites[T] = macro SoyMacroImpl.typedWritesImpl[T]
```

Comment:
Btw I tried making this optional with a default value of None, but default arguments are not allowed for macros.